crates/pallet-domains/src/lib.rs

```diff
@@ -516,16 +515,19 @@ impl<T: Config> Pallet<T> {
         OldestReceiptNumber::<T>::put::<T::BlockNumber>(Zero::zero());
     }
 
+    fn receipts_are_consecutive(
+        receipts: &[ExecutionReceipt<T::BlockNumber, T::Hash, T::DomainHash>],
+    ) -> bool {
+        (0..receipts.len() - 1)
+            .all(|i| receipts[i].primary_number + One::one() == receipts[i + 1].primary_number)
```

Comment:
Looks like you want https://doc.rust-lang.org/stable/std/primitive.slice.html#method.array_windows here for better readability and avoiding managing indexes entirely.

Reply:
Thanks for the clue, will have a look later. @NingLin-P You could also help if you have time.